src/types/component.go

```diff
@@ -137,14 +137,36 @@ type ZarfComponentActionDefaults struct {
-	MaxTotalSeconds *int     `json:"maxTotalSeconds,omitempty" jsonschema:"description=Timeout in seconds for the command (default to 0, no timeout)"`
-	MaxRetries      *int     `json:"maxRetries,omitempty" jsonschema:"description=Retry the command if it fails up to given number of times (default 0)"`
-	Dir             *string  `json:"dir,omitempty" jsonschema:"description=The working directory to run the command in (default is CWD)"`
-	Env             []string `json:"env,omitempty" jsonschema:"description=Additional environment variables to set for the command"`
-	Cmd             string   `json:"cmd,omitempty" jsonschema:"description=The command to run"`
-	SetVariable     string   `json:"setVariable,omitempty" jsonschema:"description=The name of a variable to update with the output of the command. This variable will be available to all remaining actions and components in the package.,pattern=^[A-Z0-9_]+$"`
-	Description     string   `json:"description,omitempty" jsonschema:"description=Description of the action to be displayed during package execution instead of the command"`
+	Mute            *bool                    `json:"mute,omitempty" jsonschema:"description=Hide the output of the command during package deployment (default false)"`
+	MaxTotalSeconds *int                     `json:"maxTotalSeconds,omitempty" jsonschema:"description=Timeout in seconds for the command (default to 0, no timeout)"`
+	MaxRetries      *int                     `json:"maxRetries,omitempty" jsonschema:"description=Retry the command if it fails up to given number of times (default 0)"`
+	Dir             *string                  `json:"dir,omitempty" jsonschema:"description=The working directory to run the command in (default is CWD)"`
+	Env             []string                 `json:"env,omitempty" jsonschema:"description=Additional environment variables to set for the command"`
+	Cmd             string                   `json:"cmd,omitempty" jsonschema:"description=The command to run. Must specify either cmd or wait for the action to do anything."`
+	SetVariable     string                   `json:"setVariable,omitempty" jsonschema:"description=The name of a variable to update with the output of the command. This variable will be available to all remaining actions and components in the package.,pattern=^[A-Z0-9_]+$"`
+	Description     string                   `json:"description,omitempty" jsonschema:"description=Description of the action to be displayed during package execution instead of the command"`
+	Wait            *ZarfComponentActionWait `json:"wait,omitempty" jsonschema:"description=Wait for a condition to be met before continuing. Must specify either cmd or wait for the action to do anything. If both are specified, the command will be run first, then the wait condition will be checked. See the 'zarf tools wait-for' command for more info."`
```

Comment:
I thought we talked about only allowing one or the other?

Reply:
Yes it says it in the docs, JSON schema and has a validator to run on `package create`, but our current setup doesn't support JSON Schema One-of (I tried it)

Reply:
This text says: "If both are specified, the command will be run first" which cannot happen since they cannot both be specified then.